backend/app/controllers/spree/admin/stock_items_controller.rb

```diff
@@ -11,41 +18,82 @@ def update
       end
 
       def create
-        variant = Variant.find(params[:variant_id])
-        stock_location = StockLocation.find(params[:stock_location_id])
         stock_movement = stock_location.stock_movements.build(stock_movement_params)
         stock_movement.stock_item = stock_location.set_up_stock_item(variant)
 
         if stock_movement.save
           flash[:success] = flash_message_for(stock_movement, :successfully_created)
```

Comment:
We should set `flash[:success]` specifically for `html` format.
Currently if we create a stock_movement using js request, the flash message will be shown. Then if we refresh the page or navigate to some other page, then the same flash message will be shown again.